evm/src/main/java/com/horizen/evm/LibEvm.java

```diff
@@ -0,0 +1,154 @@
+        invoke(method, args, Void.class);
+    }
+
+    private static void invoke(String method) throws Exception {
+        invoke(method, null, Void.class);
+    }
+
+    public static void openMemoryDB() throws Exception {
+        invoke("OpenMemoryDB");
+    }
+
+    public static void openLevelDB(String path) throws Exception {
+        invoke("OpenLevelDB", new LevelDBParams(path));
+    }
+
+    public static void closeDatabase() throws Exception {
+        invoke("CloseDatabase");
+    }
+
+    public static int stateOpen(byte[] root) throws Exception {
+        return invoke("StateOpen", new OpenStateParams(root), int.class);
+    }
+
+    public static void stateClose(int handle) throws Exception {
+        invoke("StateClose", new HandleParams(handle));
+    }
+
+    public static byte[] stateIntermediateRoot(int handle) throws Exception {
+        return invoke("StateIntermediateRoot", new HandleParams(handle), Hash.class).toBytes();
+    }
+
+    public static byte[] stateCommit(int handle) throws Exception {
+        return invoke("StateCommit", new HandleParams(handle), Hash.class).toBytes();
+    }
+
+    public static BigInteger stateGetBalance(int handle, byte[] address) throws Exception {
+        return invoke("StateGetBalance", new AccountParams(handle, address), BigInteger.class);
+    }
+
+    public static void stateAddBalance(int handle, byte[] address, BigInteger amount) throws Exception {
+        invoke("StateAddBalance", new BalanceParams(handle, address, amount));
+    }
+
+    public static void stateSubBalance(int handle, byte[] address, BigInteger amount) throws Exception {
+        invoke("StateSubBalance", new BalanceParams(handle, address, amount));
+    }
+
+    public static void stateSetBalance(int handle, byte[] address, BigInteger amount) throws Exception {
+        invoke("StateSetBalance", new BalanceParams(handle, address, amount));
+    }
+
+    public static long stateGetNonce(int handle, byte[] address) throws Exception {
+        return invoke("StateGetNonce", new AccountParams(handle, address), long.class);
+    }
+
+    public static void stateSetNonce(int handle, byte[] address, long nonce) throws Exception {
+        invoke("StateSetNonce", new NonceParams(handle, address, nonce));
+    }
+
+    public static byte[] stateGetCodeHash(int handle, byte[] address) throws Exception {
+        return invoke("StateGetCodeHash", new AccountParams(handle, address), Hash.class).toBytes();
+    }
+
+    public static EvmResult evmExecute(int handle, byte[] from, byte[] to, BigInteger value, byte[] input)
+        throws Exception {
+        var cfg = new EvmParams.EvmConfig();
+        cfg.origin = Address.FromBytes(from);
+        cfg.value = value;
+        cfg.gasLimit = 100000;
+        return invoke("EvmExecute", new EvmParams(handle, cfg, to, input), EvmResult.class);
+    }
+
+    public static EvmApplyResult evmApply(int handle, byte[] from, byte[] to, BigInteger value, byte[] input)
```

Comment:
This is an alternative method to call the EVM which uses more of the code-path in GETH when applying a block. It has more validation and also handles gas fees (Intrinsic Gas + moving gas from the caller balance to the coinbase account). Which one we're going to use, or if we need something a bit different depends on how exactly we want to handle gas fees.